ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/AtomicActionRecoveryModule.java

```diff
@@ -253,4 +257,6 @@ public boolean hasWorkLeftToDo() {
 
     private boolean hasWorkLeftToDo;
 
+    private boolean problemDuringRecovery;
```

Comment:
Please can you explain the advantage of a new property rather than just setting hasWorkLeftToDo in these places? Our API method is called hasWorkLeftToDo() so how does the implementation benefit from having this new variable used? A consequence is that the hasWorkLeftToDo() method seems a bit more complicated now too because it is considering problemDuringRecovery as well as the previous hasWorkLeftToDo

Reply:
I think that a dedicated variable that tracks problems is a good separation of concerns: `hasWorkLeftToDo` will track left-over transactions while `problemDuringRecovery` will track exceptions/problems. This is also in line with `XARecoveryModule` (and my other PR #2336)

Reply:
> A consequence is that the hasWorkLeftToDo() method seems a bit more complicated now too because it is considering problemDuringRecovery as well as the previous hasWorkLeftToDo

With my modifications, `hasWorkLeftToDo()` will still return when there is "work left to do" but that situation will be associated to having either left-over tranasctions or problems during recovery

Reply:
OK, that is good for me - thank you

Reply:
Thanks Tom